test/controllers/v1/profiles_controller_test.rb

```diff
@@ -11,28 +11,65 @@ class ProfilesControllerTest < ActionDispatch::IntegrationTest
       profiles(:one).update! account: accounts(:test)
     end
 
-    test 'tailoring_file with a canonical profile returns no content' do
-      profiles(:one).update! rules: [rules(:one)]
-      get tailoring_file_v1_profile_url(profiles(:one).id)
-      assert_response :no_content
-    end
+    class TailoringFileTest < ProfilesControllerTest
+      test 'tailoring_file with a canonical profile returns no content' do
+        profiles(:one).update! rules: [rules(:one)]
+        get tailoring_file_v1_profile_url(profiles(:one).id)
+        assert_response :no_content
+      end
+
+      test 'tailoring_file with a noncanonical profile matching its parent'\
+        'returns no content' do
+        profiles(:two).update!(rules: [rules(:one)])
+        profiles(:one).update!(parent_profile: profiles(:two),
+                               rules: [rules(:one)])
+        get tailoring_file_v1_profile_url(profiles(:one).id)
+        assert_response :no_content
+      end
 
-    test 'tailoring_file with a noncanonical profile matching its parent'\
-      'returns no content' do
-      profiles(:two).update!(rules: [rules(:one)])
-      profiles(:one).update!(parent_profile: profiles(:two),
-                             rules: [rules(:one)])
-      get tailoring_file_v1_profile_url(profiles(:one).id)
-      assert_response :no_content
+      test 'tailoring_file with a noncanonical profile '\
+           'returns tailoring file' do
+        profiles(:two).update!(rules: [rules(:one), rules(:two)])
+        profiles(:one).update!(parent_profile: profiles(:two),
+                               rules: [rules(:one)])
+        get tailoring_file_v1_profile_url(profiles(:one).id)
+        assert_response :success
+        assert_equal Mime[:xml].to_s, @response.content_type
+      end
     end
 
-    test 'tailoring_file with a noncanonical profile returns tailoring file' do
-      profiles(:two).update!(rules: [rules(:one), rules(:two)])
-      profiles(:one).update!(parent_profile: profiles(:two),
-                             rules: [rules(:one)])
-      get tailoring_file_v1_profile_url(profiles(:one).id)
-      assert_response :success
-      assert_equal Mime[:xml].to_s, @response.content_type
+    class DestroyTest < ProfilesControllerTest
+      require 'sidekiq/testing'
```

Comment:
I tried removing it, there was no issue with it - I think for this to be effective you need to associate the profile to some hosts so that some DeleteHost jobs get actually enqueued

Reply:
Odd - I get errors when I remove it. Are you running the tests on their own or the whole suite?

```
$ d-c exec -e TEST=$TEST reports-consumer bundle exec rake test TEST=test/controllers/v1/profiles_controller_test.rb
Started with run options --seed 31260

DEPRECATION WARNING: 'alias' argument deprecated, prefer aliases: [..] since scoped_search 4.0.0 (called from <class:Rule> at /app/app/models/rule.rb:9)---] 0% Time: 00:00:00,  ETA: ??:??:??
DEPRECATION WARNING: 'alias' argument deprecated, prefer aliases: [..] since scoped_search 4.0.0 (called from <class:Rule> at /app/app/models/rule.rb:10)
DEPRECATION WARNING: 'alias' argument deprecated, prefer aliases: [..] since scoped_search 4.0.0 (called from <class:Benchmark> at /app/app/models/xccdf/benchmark.rb:10)
DEPRECATION WARNING: 'alias' argument deprecated, prefer aliases: [..] since scoped_search 4.0.0 (called from <class:Benchmark> at /app/app/models/xccdf/benchmark.rb:12)
ERROR["test_v1_destroy_an_existing,_accessible_profile", #<Minitest::Reporters::Suite:0x000055c08dd0be18 @name="V1::ProfilesControllerTest::DestroyTest">, 0.367351297987625]
 test_v1_destroy_an_existing,_accessible_profile#V1::ProfilesControllerTest::DestroyTest (0.37s)
Minitest::UnexpectedError:         Redis::CannotConnectError: Error connecting to Redis on 127.0.0.1:6379 (Errno::ECONNREFUSED)
            app/models/concerns/profile_policy_association.rb:40:in `destroy_policy_test_results'
            app/controllers/v1/profiles_controller.rb:17:in `destroy'
            test/controllers/v1/profiles_controller_test.rb:53:in `block in <class:DestroyTest>'

ERROR["test_destroy_a_non-existant_profile", #<Minitest::Reporters::Suite:0x000055c08e2920d0 @name="V1::ProfilesControllerTest::DestroyTest">, 0.38759273797040805]
 test_destroy_a_non-existant_profile#V1::ProfilesControllerTest::DestroyTest (0.39s)
Minitest::UnexpectedError:         Redis::CannotConnectError: Error connecting to Redis on 127.0.0.1:6379 (Errno::ECONNREFUSED)
            app/models/concerns/profile_policy_association.rb:40:in `destroy_policy_test_results'
            test/controllers/v1/profiles_controller_test.rb:62:in `block in <class:DestroyTest>'

ERROR["test_destroy_an_existing,_accessible_profile", #<Minitest::Reporters::Suite:0x000055c08f67ff48 @name="V1::ProfilesControllerTest::DestroyTest">, 0.4133574939914979]
 test_destroy_an_existing,_accessible_profile#V1::ProfilesControllerTest::DestroyTest (0.41s)
Minitest::UnexpectedError:         Redis::CannotConnectError: Error connecting to Redis on 127.0.0.1:6379 (Errno::ECONNREFUSED)
            app/models/concerns/profile_policy_association.rb:40:in `destroy_policy_test_results'
            app/controllers/v1/profiles_controller.rb:17:in `destroy'
            test/controllers/v1/profiles_controller_test.rb:44:in `block in <class:DestroyTest>'

  7/7: [=================================================================================================================================================] 100% Time: 00:00:00, Time: 00:00:00

Finished in 0.57739s
7 tests, 5 assertions, 0 failures, 3 errors, 0 skips
Coverage report generated for RSpec to /app/coverage. 434 / 2903 LOC (14.95%) covered.
```

Reply:
I was running the test on its own